packages/customize-widgets/src/store/selectors.js

```diff
@@ -35,6 +35,5 @@ export function isInserterOpened( state ) {
  * @return {Object} The root client ID and index to insert at.
  */
 export function __experimentalGetInsertionPoint( state ) {
-	const { rootClientId, insertionIndex } = state.blockInserterPanel;
-	return { rootClientId, insertionIndex };
```

Comment:
Agree. This is a bit confusing way to handle the state. I was just trying to keep changes related to the issue here, but I am also happy to work on those improvements.

The same state is responsible for opening the block inserter - `setIsInserterOpened( true )`. How do you suggest handling a `truthy` value? The `insertionPoint` data isn't required for opening the inserter.

Reply:
For the `__experimentalGetInsertionPoint` selector, it doesn't matter at all whether the state value is `true` or `false`. The `const { rootClientId } = state` destructuring always yields the same `undefined` value.

But patching the reducers is not as easy as I originally thought, because there's also the `isInserterOpened` selectors that cares about the boolean value.

We could do this:
```js
const EMPTY_INSERTION_POINT = {
  rootClientId: undefined,
  insertionIndex: undefined,
};
function __experimentalGetInsertionPoint( state ) {
  if ( typeof state == 'boolean' ) {
    return EMPTY_INSERTION_POINT;
  }
  return state;
}
```
That gives the selector a consistent return type, the same as it had before this PR, and it should be easy to implement.

Reply:
Sounds good. I'll update the PR.

I think in the future, we could split the state in two - `isInserterOpen` and `blockInsertionPoint`. The latter can live in the block editor package, and then we don't have to inject the `__experimentalSetIsInserterOpened` callback via settings.

Reply:
Updated in 139d7a8.